src/milia/api/io.clj

```diff
@@ -86,21 +86,36 @@
 
 (defn parse-binary-response
   "Parse binary response by writing into a temp file and returning the path."
-  [body filename]
+  [body filename & {:keys [url http-options]}]
   (let [tempfile (File/createTempFile filename "")
         path (str (.getAbsolutePath tempfile))
-        ^File file (clojure.java.io/file path)]
+        ^File file (clojure.java.io/file path)
+        ;; Stream the http-request to avoid out of memory errors when the data
+        ;; to copy is large
+        streamed-http-request
```

Comment:
We could destructure `streamed-http-request` here and use `status` and `body` in lines `101`, `102` and `107`.